src/state/url-sync-mixin.ts

```diff
@@ -13,6 +13,12 @@ import { Constructor } from "../types";
 
 const DEBUG = false;
 
+export const historyPromise: { promise?: Promise<void> } = {
```

Comment:
No need to make it an object. Exports are not "copied" into the scope of the other module and you can just override it's definition.